app/src/main/java/com/amaze/filemanager/filesystem/Operations.java

```diff
@@ -441,6 +441,63 @@ public static void rename(
 
       private final DataUtils dataUtils = DataUtils.getInstance();
 
+      private final Boolean isCaseSensitiveRename =
+              !oldFile.getSimpleName().equals(newFile.getSimpleName()) &&
+              oldFile.getSimpleName().equalsIgnoreCase(newFile.getSimpleName());
+
+      // random string that is appended to file to prevent name collision
+      private static final String TEMP_FILE_EXT = "qwerty";
```

Comment:
This is too less random. Please randomize a bit more. Plus, suggested to put this at the beginning of the filename instead.

Reply:
Putting it at the beginning of filename requires paths which is only available for API level 26 or higher. Doing it manually by string manipulation is risky. I have changed the temporary constant to a 32-characters string generated from `!@#$%a-zA-z0-9`.